edge-modules/Relayer/Settings.cs

```diff
@@ -24,24 +26,40 @@ public class Settings
 
                 return new Settings(
                     configuration.GetValue("transportType", TransportType.Amqp_Tcp_Only),
+                    configuration.GetValue("inputName", "input1"),
                     configuration.GetValue("outputName", "output1"));
             });
 
         Settings(
             TransportType transportType,
+            string inputName,
             string outputName)
         {
-            this.TransportType = Preconditions.CheckNotNull(transportType);
+            this.InputName = Preconditions.CheckNonWhiteSpace(inputName, nameof(inputName));
             this.OutputName = Preconditions.CheckNonWhiteSpace(outputName, nameof(outputName));
+            this.TransportType = transportType;
```

Comment:
Is there a Check if the enum is valid within a set of `TransportType`?

Reply:
configuration.GetValue will throw an error if invalid

Reply:
Currently all values in TransportType enum are valid here.  Also afaik i don't know there is any method in Precondition class to verify enum.